xds/src/main/java/com/linecorp/armeria/xds/XdsEndpointGroup.java

```diff
@@ -90,4 +151,38 @@ protected void doCloseAsync(CompletableFuture<?> future) {
         safeCloseable.close();
         super.doCloseAsync(future);
     }
+
+    private void setEndpoints(EndpointSnapshot endpointSnapshot) {
+        setEndpoints(convertEndpoints(endpointSnapshot.xdsResource().resource()));
+    }
+
+    private static Struct filterMetadata(ClusterSnapshot clusterSnapshot) {
+        final VirtualHost virtualHost = clusterSnapshot.virtualHost();
+        assert virtualHost != null;
+        final Route routes = virtualHost.getRoutes(0);
```

Comment:
Question) Would just using `clusterSnapshot.route()` directly make more sense?
There is no guarantee that the `route` here corresponds to `clusterSnapshot`

Reply:
That's a good suggestion!